### PR TITLE
DIR-7400: API documentation for conversations / messaging functionality + dependency updates

### PR DESCRIPTION
**Pull request by jakebills1**
This adds a new section with documentation for 4 new API endpoints, viewable locally at http://localhost:4567/#messaging.
In addition, since I was not able to run this app on my machine anymore, this updates the dependencies to their latest versions where possible. `middleman-sprockets` was no longer necessary to process Sass files with the latest `middleman` version, so it was removed.

Since this does not only add to the documentation, please test locally by running:
```shell
bundle install
bundle exec middleman server
```
and viewing the app at http://localhost:4567/

### CODE REVIEW

File: source/index.html.md [hunk 1]
```diff
@@ -3510,6 +3510,205 @@ Parameter | Description
...
+active (optional) | Filter by active conversations, defined as conversations associated with confirmed bookings that have a checkout date no more than 2 days past.
+
+Example with all optional parameters
+`/api/public/990/conversations?_limit=50&_offset=50&active=true&created_after=2021-06-17`
+
+## Get a Conversation
+
+```shell
+curl "https://staging.getdirect.io/api/public/<org_id>/conversations/<id>"
+  -h "authorization: token your_api_key"
```
Comment (cariperi) [inline source/index.html.md:3582]:
```suggestion
  -H "Authorization: Token your_api_key"
```

File: source/index.html.md [hunk 2]
```diff
@@ -3510,6 +3510,205 @@ Parameter | Description
...
+  -h "accept: application/vnd.direct.v1"
```
Comment (cariperi) [inline source/index.html.md:3583]:
```suggestion
  -H "Accept: application/vnd.direct.v1"
```

File: source/index.html.md [hunk 3]
```diff
@@ -3510,6 +3510,205 @@ Parameter | Description
...
+created_after (optional) | Filter by conversations created after a specific date.
+active (optional) | Filter by active conversations, defined as conversations associated with confirmed bookings that have a checkout date no more than 2 days past.
+
+Example with all optional parameters
+`/api/public/990/conversations?_limit=50&_offset=50&active=true&created_after=2021-06-17`
+
+## Get a Conversation
+
+```shell
+curl "https://staging.getdirect.io/api/public/<org_id>/conversations/<id>"
```
Comment (cariperi) [inline source/index.html.md:3581]:
```suggestion
curl "https://staging.getdirect.io/api/public/<ORG_ID>/conversations/<ID>"
```

File: source/index.html.md [hunk 4]
```diff
@@ -3510,6 +3510,205 @@ Parameter | Description
...
+    "original_message": null,
+    "channel_data": {},
+    "organization_id": 890,
+    "external_id": null
+}
+```
+
+This endpoint creates a message in a particular conversation.
+
+The 'body' param in the request body will have all HTML tags stripped out and then surrounded by p tags before being saved.
```
Comment (cariperi) [inline source/index.html.md:3708]:
```suggestion
The 'body' param in the request body will have all HTML tags stripped out and then be surrounded by p tags before being saved.
```

File: source/index.html.md [hunk 5]
```diff
@@ -3510,6 +3510,205 @@ Parameter | Description
...
+
+`GET /conversations/<CONVERSATION_ID>/messages`
+
+### URL Parameters
+
+Parameter | Description
+--------- | -----------
+ID | The ID of the conversation to filter messages by
+
+## Create a Message
```
Comment (cariperi) [inline source/index.html.md:3674]: Are we intentionally leaving out subsections with the HTTP Request and Request Parameters for this request section?

It looks like Create requests for Reviews and Reservations do have this documented, but I think it's clear enough from the cURL example if you want to omit.

**Review by cariperi — CHANGES_REQUESTED** [review]
Just a couple of small suggestions, mostly formatting, and a question before approving.

I was able to run the specified commands and run this branch locally on my machine, so the changes to Gemfile / config look good to me.

**Comment by jakebills1** [comment]
@cariperi thanks for the review, the case changes in the curl examples were all unintentional, good catch. Also added the HTTP request documentation that was missing

**Review by cariperi — APPROVED** [review]
Thanks for updating - looks great! I have a ticket coming up to work on these docs, so was good to clone down and see how it works.

Record of page loading locally with config changes from this PR:
image

**Pull request merged**